mne/encoding/feature.py

```diff
@@ -0,0 +1,298 @@
+        """Binarize events and return as continuous data.
+
+        Parameters
+        ----------
+        event_ixs : array, shape (n_events)
+            The event indices or onsets in seconds. `If self.sfreq == 1`,
+            then these are assumed to be indices. If `self.sfreq != 1`, these
+            should be onsets in seconds and will be multiplied by `sfreq`.
+        event_ids : array-like, shape (n_events) | None
+            The event ID for each event. If None, all events are assumed to be
+            of same type.
+        event_dict : dict
+            A dictionary of (event_id: event_id_name) pairs. Defines a string
+            name for each event type.
+        covariates : array, shape (n_events, n_covariates) | None
+            Optional covariates (e.g., continuous values) for each event.
+        covariate_names : array, shape (n_covariates,) | None
+            Optional covariate names.
+
+        Returns
+        -------
+        events_continuous : array, shape (n_times, n_unique_events)
+            A binary array with one column for each unique event type. 1s
+            in the array correspond to event onsets for that row.
+        """
+        if event_ixs.ndim > 1:
+            raise ValueError("events must be shape (n_events,),"
+                             " found shape %s" % str(event_ixs.shape))
+        event_ids = np.ones_like(event_ixs) if event_ids is None else event_ids
+        unique_event_types = np.unique(event_ids)
+        covs, cov_names = _check_covariates(event_ixs, covariates,
+                                            covariate_names)
+        n_covs = covs.shape[-1]
+
+        # Create names for event types
+        if event_dict is None:
+            event_dict = dict(('event_%s' % ii, ii)
+                              for ii in unique_event_types)
+        ev_dict_rev = dict((value, key) for key, value in event_dict.items())
+
+        # Turn event_ixs from seconds to indices in case sfreq != 1
+        event_ixs = (event_ixs * self.sfreq).astype(int)
+
+        # Iterate through event types and create columns of event onsets.
+        events_continuous = np.zeros([self.n_times,
+                                      len(unique_event_types) + n_covs])
+        event_names = []
+        for ii, ev_type in enumerate(unique_event_types):
+            msk_events = event_ids == ev_type
+            i_ev = event_ixs[msk_events]
+            events_continuous[i_ev, ii] = 1
+
+            # Handle event names
+            event_names.append(ev_dict_rev[ev_type])
+        if n_covs > 0:
+            for iev, icov in zip(event_ixs, covs):
+                events_continuous[iev, -n_covs:] = icov
+            event_names = event_names + cov_names
+            events_continuous = sparse.csr_matrix(events_continuous)
+
+        self.names_ = event_names
+        self.unique_event_types_ = unique_event_types
+        return events_continuous
```

Comment:
it's got some of the same functionality, but that function both created binary continuous events, and created time delays. This PR is partially about modularizing that functionality into specific classes, thus 2 separate classes for both of those operations (binarizing and delaying)